libs/datasets/combined_datasets.py

```diff
@@ -169,12 +171,16 @@ def build_combined_dataset_from_sources(
 
     # Build feature columns from feature_definition_config.
     data = pd.DataFrame({})
+    log = structlog.get_logger()
     for field, data_source_classes in feature_definition_config.items():
         for data_source_cls in data_source_classes:
             dataset = intermediate_datasets[data_source_cls]
-
             data = dataset_utils.fill_fields_with_data_source(
-                data, dataset.data, target_dataset_cls.INDEX_FIELDS, [field]
+                log.bind(dataset_name=data_source_cls.SOURCE_NAME, field=field),
```

Comment:
I'm not sure I understand why structlog is being used here? (mostly because I'm just not familiar with it)

Reply:
> LGTM! It would be good to run the api generation locally on recent model output to make sure that the results match before and after this change (at least on the state level)

I ran 
`python ./run.py deploy-states-api -i api-results-371/ -o api-results-371-output --summary-output api-results-371-output-states` and again with master checked
` python ./run.py deploy-states-api -i api-results-371/ -o api-results-371-output-master --summary-output api-results-371-output-states-master`
and `diff -r api-results-371-output-states api-results-371-output-states-master` found no differences.